hdf/src/hcomp.c

```diff
@@ -1057,8 +1054,7 @@ HCIstaccess(accrec_t *access_rec, int16 acc_mode)
 
 done:
     if (ret_value == FAIL) { /* Error condition cleanup */
-        if (info != NULL)
-            HDfree(info);
+        free(info);
```

Comment:
info is a local variable alias for access_rec->special_info.  freeing info will free access_rec->special_info and then access_rec->special_info will be invalid. access_rec->special_info should be freed and set to NULL. 
Maybe the alias should not be used for the free elsewhere as well. (see lines 1424-1425)

Reply:
In this case, HCIstaccess, it is ok because access_rec->special_info is being created/initialized and, when errors occur, whatever there for access_rec->special_info already will be freed via info.
However, the better way is to use info the whole time and only make access_rec->special_info when successs.